statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitor.java

```diff
@@ -0,0 +1,171 @@
+package com.avast.metrics.statsd;
+
+import com.avast.metrics.TimerPairImpl;
+import com.avast.metrics.api.Metric;
+import com.avast.metrics.api.Monitor;
+import com.avast.metrics.api.Naming;
+import com.avast.metrics.api.TimerPair;
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"WeakerAccess", "OptionalUsedAsFieldOrParameterType", "unused"})
+public class StatsDMetricsMonitor implements Monitor {
+    protected final StatsDClient client;
+    protected final String domain;
+    protected final List<String> names = new ArrayList<>();
+    protected final Naming naming;
+    protected final ScheduledExecutorService scheduler;
+    protected final Duration gaugeSendPeriod;
+
+    protected final Map<String, ScheduledFuture<?>> gauges = new HashMap<>();
+
+    public StatsDMetricsMonitor(String host, int port, String domain, final Naming naming, final Duration gaugeSendPeriod, final ScheduledExecutorService scheduler) {
```

Comment:
The `domain` word is not used in StatsD context - please use `prefix` instead - it's the same name for this concept that is used in the underlying library.